test/test_compress_bound.cc

```diff
@@ -0,0 +1,60 @@
+/* test_compress_bound.cc - Test compressBound() with small buffers */
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+#else
+#  include "zlib-ng.h"
+#endif
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "test_shared.h"
+
+#include <gtest/gtest.h>
+
+#define MAX_LENGTH (32)
+
+class compress_bound_variant : public testing::TestWithParam<z_size_t> {
+public:
+    void estimate(z_size_t level) {
+        int estimate_len = 0;
+        uint8_t *uncompressed = NULL;
+        uint8_t dest[128];
+        int err;
+
+        uncompressed = (uint8_t *)malloc(MAX_LENGTH);
+        ASSERT_TRUE(uncompressed != NULL);
+
+        /* buffer with values for worst case compression */
+        for (int j = 0; j < MAX_LENGTH; j++) {
+            uncompressed[j] = (uint8_t)j;
+        }
+
+        for (int32_t i = 0; i < MAX_LENGTH; i++) {
+            /* calculate actual output length */
+            estimate_len = PREFIX(compressBound)(i);
+
+            if (dest != NULL) {
+                z_size_t dest_len;
```

Comment:
`dest_len` should be initialized to output buffer length, otherwise compress2() will not know how many bytes it can write to output buffer before it overflows. Returned value is how many bytes was actually used, so the parameter is read/write.

Reply:
Fixed.